agasc/agasc.py

```diff
@@ -1,42 +1,178 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import os
 from pathlib import Path
+import warnings
+from contextlib import ContextDecorator
 
 import numpy as np
 import numexpr
 import tables
-import warnings
 
 from Chandra.Time import DateTime
 from astropy.table import Table, Column
-from ska_helpers.utils import LazyDict
+from ska_helpers.utils import lru_cache_timed
+
+__all__ = ['sphere_dist', 'get_agasc_cone', 'get_star', 'get_stars',
+           'MAG_CATID_SUPPLEMENT', 'BAD_CLASS_SUPPLEMENT', 'get_supplement_table',
+           'disable_supplement']
+
+DISABLE_SUPPLEMENT_ENV = 'AGASC_DISABLE_SUPPLEMENT'
+MAG_CATID_SUPPLEMENT = 100
+BAD_CLASS_SUPPLEMENT = 100
+
+RA_DECS_CACHE = {}
+COMMON_DOC = """\
+If ``use_supplement`` is ``True`` and there is no ``AGASC_DISABLE_SUPPLEMENT``
+    environment variable set, then stars with available mag estimates or bad
+    star entries in the AGASC supplement are updated in-place in the output
+    ``stars`` Table:
+
+    - ``MAG_ACA`` and ``MAG_ACA_ERR`` are set according to the supplement.
+    - ``MAG_CATID`` (mag catalog ID) is set to ``MAG_CATID_SUPPLEMENT`` (100).
+    - If ``COLOR1`` is 0.7 or 1.5 then it is changed to 0.69 or 1.49 respectively.
+    - If ``AGASC_ID`` is in the supplement bad stars table then CLASS is set to
+      ``100 + <source>``, where ``<source>`` is the bad star source identifier.
```

Comment:
I don't like having hardwired numbers in the code. You have added three. I can understand the replacements for 0.7 and 1.5 because the idea is that now they would not be the old hardwired numbers and you can still use COLOR1 as a proxy for large uncertainty. Not pretty but ok.

I do not understand why add the hardwired source. Couldn't you have a "bad" column?  This kind convention (100 + source) is not good style, so why do it in the first place if it is not necessary? It is not needed for backward compatibility, is it? and having a "bad" column would be explicit.

Reply:
I understand your concerns. This code falls in the category of "perfect is the enemy of good".

Using `CLASS` instead of a new column is needed for backward compatibility. Proseco, starcheck, and MATLAB tools (SAUSAGE) use `CLASS` as a way of identifying whether a star might be acceptable. For instance `CLASS` is in the star information hover-over now in starcheck and impacts how stars are plotted. So a CLASS of 100 or more will ensure that existing code will recognize that star as (generally speaking) not accepted for star selection. 

I am perfectly happen to just use `BAD_CLASS_SUPPLEMENT` always, so unless @jeanconn objects I'll do that.

Another driver here is that I might like to clean up `proseco` so it doesn't specifically look at the `bad` HDF5 table, but instead just defers to `agasc` to tell that a star is bad based on CLASS. This probably already happens for free with no code change.

Reply:
Fixed, to the extent practical.